client/allocdir/alloc_dir.go

```diff
@@ -64,14 +91,20 @@ type AllocDirFS interface {
 	ChangeEvents(path string, curOffset int64, t *tomb.Tomb) (*watch.FileChanges, error)
 }
 
-func NewAllocDir(allocDir string) *AllocDir {
-	d := &AllocDir{AllocDir: allocDir, TaskDirs: make(map[string]string)}
+func NewAllocDir(allocDir string, maxSize int) *AllocDir {
+	d := &AllocDir{
+		AllocDir: allocDir,
+		maxSize:  maxSize,
+		TaskDirs: make(map[string]string),
+		watchCh:  make(chan struct{}),
```

Comment:
It may be worth replicating the constants here as exposed fields and the constructer can set the values to the constants and the rest of the code refers to the instance fields.

Then the tests can set the values to something much shorter and we can add tests to see that it checks the folder quickly and first and then slows down etc. We can use the file ModifyTime to check this